File: Baekjoon/10828.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const [n, ...orders] = fs.readFileSync(filePath).toString().trim().split("\n");
let stack = [];
let result = [];

orders.forEach((ele) => {
  switch (ele) {
    case "pop":
      result.push(stack.length > 0 ? stack.pop() : -1);
      break;
    case "size":
      result.push(stack.length);
      break;
    case "empty":
      result.push(stack.length > 0 ? 0 : 1);
      break;
    case "top":
      result.push(stack.length > 0 ? stack[stack.length - 1] : -1);
      break;
    default:
      let [order, num] = ele.split(" ");
      stack.push(num);
      break;
  }
});

console.log(result.join("\n"));

File: python/Baekjoon/10828.py
import sys

n = int(sys.stdin.readline().strip())
stack = []

for _ in range(n):
    command = sys.stdin.readline().strip().split()
    if command[0] == "push": stack.append(command[1])
    else:
        stackSize = len(stack)
        if command[0] == "size": print(stackSize)
        elif stackSize == 0:
            if command[0] == "empty": print(1)
            else: print(-1)
        else:
            if command[0] == "empty": print(0)
            elif command[0] == "pop": print(stack.pop())
            else: print(stack[-1])